abm_sketch.py

```diff
@@ -353,14 +427,14 @@ def make_points(cluster, size, y_dist, x_dist):
     return list(zip(xs, ys, [cluster] * size))
 
 
-def make_population(y_pos_dist, cluster_x_dists, cluster_sizes):
+def make_population(y_pos_dist, cluster_x_dists, cluster_sizes, pop_class=Population):
```

Comment:
It seems like all this constructing logic should be in the init method of the population class!